'Minisat::OutOfMemoryException' during build

During build, `/build/source/inst/bin/bsc -stdlib-names -bdir /build/source/build/bsvlib -p . -vsearch /build/source/build/bsvlib -no-use-prelude Prelude.bs` is invoked.

It fails

```
terminate called after throwing an instance of 'Minisat::OutOfMemoryException'
make[3]: *** [Makefile:22: /build/source/build/bsvlib/Prelude.bo] Aborted (core dumped)
make[3]: Leaving directory '/build/source/src/Libraries/Base1'
make[2]: *** [Makefile:31: build] Error 2
make[2]: Leaving directory '/build/source/src/Libraries'
make[1]: *** [Makefile:16: install] Error 2
make[1]: Leaving directory '/build/source/src'
make: *** [Makefile:24: install] Error 2
```

Full Log: https://termbin.com/0nb3

Note the machine has ~60GiB of RAM, so I hope it's not really not enough memory ;-)

I have same problem with debian (in docker)

## Environment

hostmachine:
```
uname -a
Linux ryzen 5.5.2-gentoo #1 SMP PREEMPT Wed Feb 5 21:46:42 CST 2020 x86_64 AMD Ryzen 9 3900X 12-Core Processor AuthenticAMD GNU/Linux
```
docker with debian:bullseye 

## The Dockerfile:

```Dockerfile
FROM debian:bullseye as installer
LABEL maintainer="Huang Rui"
# Update and install software
RUN DEBIAN_FRONTEND=noninteractive apt-get update -qq && \
    DEBIAN_FRONTEND=noninteractive apt-get install -yq \
        ghc \
        libghc-regex-compat-dev \
        libghc-syb-dev \
        libghc-old-time-dev \
        libfontconfig1-dev \
        libxft-dev \
        build-essential \
        autoconf \
        gperf \
        bison \
        flex \
        file \
        wget \
        python \
        unzip \
        rsync \
        sudo \
        git \
        curl

RUN git clone --progress --recursive https://github.com/B-Lang-org/bsc

RUN cd bsc && \
    make PREFIX=/opt/bluespec
```

## GHC version

```
Glasgow Haskell Compiler, Version 8.6.5, stage 2 booted by GHC version 8.6.5
Using binary package database: /usr/lib/ghc/package.conf.d/package.cache
package flags []
loading package database /usr/lib/ghc/package.conf.d
wired-in package ghc-prim mapped to ghc-prim-0.5.3
wired-in package integer-gmp mapped to integer-gmp-1.0.2.0
wired-in package base mapped to base-4.12.0.0
wired-in package rts mapped to rts
wired-in package template-haskell mapped to template-haskell-2.14.0.0
wired-in package ghc mapped to ghc-8.6.5
```

## GCC version
```
gcc version 9.2.1 20200123 (Debian 9.2.1-25)
```

## Error log

The command is

${PROJ}/bsc/inst/bin/bsc -stdlib-names -bdir ${PROJ}/bsc/build/bsvlib -p . -vsearch ${PROJ}/bsc/build/bsvlib -no-use-prelude Prelude.bs

The message is

```
Warning: "Prelude.bs", line 3744, column 25: (P0223)
 Definition of `kind' is not used.
Warning: "Prelude.bs", line 3747, column 25: (P0223)
 Definition of `kind' is not used.
Warning: "Prelude.bs", line 3750, column 25: (P0223)
 Definition of `kind' is not used.
Warning: "Prelude.bs", line 3980, column 8: (P0223)
 Definition of `value' is not used.
terminate called after throwing an instance of 'Minisat::OutOfMemoryException'
make[3]: Leaving directory '/bsc/src/Libraries/Base1'
make[3]: *** [Makefile:22: /bsc/build/bsvlib/Prelude.bo] Aborted (core dumped)
make[2]: *** [Makefile:31: build] Error 2
make[2]: Leaving directory '/bsc/src/Libraries'
make[1]: Leaving directory '/bsc/src'
make[1]: *** [Makefile:16: install] Error 2
make: *** [Makefile:24: install] Error 2
The command '/bin/sh -c cd bsc &&     make PREFIX=/opt/bluespec' returned a non-zero code: 2`
```

--- a/src/stp/src/sat/CryptoMinisat.cpp
+++ b/src/stp/src/sat/CryptoMinisat.cpp
@@ -34,7 +34,7 @@ namespace BEEV
     for (int i =0; i<ps.size();i++)
       v.push(MINISAT::Lit(var(ps[i]), sign(ps[i])));
 
-    s->addClause(v);
+    return s->addClause(v);
   }
 
   bool
@@ -63,7 +63,7 @@ namespace BEEV
 
   int CryptoMinisat::setVerbosity(int v)
   {
-    s->verbosity = v;
+    return s->verbosity = v;
   }
 
   int CryptoMinisat::nVars()

--- a/src/stp/src/sat/MinisatCore.cpp
+++ b/src/stp/src/sat/MinisatCore.cpp
@@ -22,7 +22,7 @@ namespace BEEV
   bool
   MinisatCore<T>::addClause(const SATSolver::vec_literals& ps) // Add a clause to the solver.
   {
-    s->addClause(ps);
+    return s->addClause(ps);
   }
 
   template <class T>
@@ -60,7 +60,7 @@ namespace BEEV
   template <class T>
   int MinisatCore<T>::setVerbosity(int v)
   {
-    s->verbosity = v;
+    return s->verbosity = v;
   }
 
     template <class T>
@@ -97,7 +97,7 @@ namespace BEEV
   template <class T>
     bool MinisatCore<T>::simplify()
   {
-    s->simplify();
+    return s->simplify();
   }
 
 

--- a/src/stp/src/sat/MinisatCore_prop.cpp
+++ b/src/stp/src/sat/MinisatCore_prop.cpp
@@ -30,7 +30,7 @@ namespace BEEV
   bool
   MinisatCore_prop<T>::addClause(const SATSolver::vec_literals& ps) // Add a clause to the solver.
   {
-    s->addClause(ps);
+    return s->addClause(ps);
   }
 
   template <class T>
@@ -68,7 +68,7 @@ namespace BEEV
   template <class T>
   int MinisatCore_prop<T>::setVerbosity(int v)
   {
-    s->verbosity = v;
+    return s->verbosity = v;
   }
 
   template <class T>

--- a/src/stp/src/sat/SimplifyingMinisat.cpp
+++ b/src/stp/src/sat/SimplifyingMinisat.cpp
@@ -17,7 +17,7 @@ namespace BEEV
   bool
   SimplifyingMinisat::addClause(const vec_literals& ps) // Add a clause to the solver.
   {
-    s->addClause(ps);
+    return s->addClause(ps);
   }
 
   bool
@@ -49,7 +49,7 @@ namespace BEEV
 
   int SimplifyingMinisat::setVerbosity(int v)
   {
-    s->verbosity = v;
+    return s->verbosity = v;
   }
 
   void SimplifyingMinisat::setSeed(int i)

--- a/src/stp/src/simplifier/simplifier.cpp
+++ b/src/stp/src/simplifier/simplifier.cpp
@@ -523,7 +523,7 @@ namespace BEEV
       return getConstantBit(n[0], i);
 
     assert(false);
-    return -1337;
+    abort();
   }
 
   ASTNode

--- a/src/stp/src/simplifier/simplifier.cpp
+++ b/src/stp/src/simplifier/simplifier.cpp
@@ -523,6 +523,7 @@ namespace BEEV
       return getConstantBit(n[0], i);
 
     assert(false);
+    return -1337;
   }
 
   ASTNode

